src/main.rs

```diff
@@ -40,14 +33,31 @@ async fn main() {
     });
 
     loop {
-        let model_clone = model.lock().await.clone();
+        let model_clone = model.clone();
+        let model = model.lock().await;
 
         canvas
-            .draw(|frame| app.render(&model_clone, frame))
+            .draw(|frame| app.render(&model, frame))
             .expect("terminal has failed to draw");
 
-        match controller.run(&model_clone, app.context_mut()) {
-            State::Continue => continue,
+        match controller.run(&model, app.context_mut()) {
+            State::Continue => {
+                thread::sleep(Duration::from_millis(100));
```

Comment:
Let's use the tokio library's sleep function instead of the thread library's one.